python-checks/src/test/resources/checks/iterMethodReturnTypeCheck.py

```diff
@@ -202,6 +206,69 @@ def __iter__(self):
         else:
             raise NotImplementedError()
 
+def noReturn01(sth) -> NoReturn:
+    ...
```

Comment:
to better explain why we're trusting the type annotation, I would replace `...` with an actual raise, e.g. `raise ValueError()`